tests/test_java.py

```diff
@@ -643,3 +644,27 @@ def test_java_different_basename(
                     os.path.basename(log_records[0].gprofiler_adapter_extra["exe"])  # type: ignore
                     == java_notjava_basename
                 )
+
+@pytest.mark.parametrize("in_container", [True])
+def test_dso_name_in_ap_profile(
```

Comment:
Test code could be shortened by use of parametrization - see e.g how `test_perf_fp_dwarf_smart` uses parametrization to run the same logic but expect different results based on the parameters. There's no need to clone the actual profiling code as done here.

Reply:
Same goes for `test_dso_name_in_perf_profile`

Reply:
👍